pkgs/development/libraries/wolfssl/default.nix

```diff
@@ -5,37 +5,63 @@
 , autoreconfHook
 , util-linux
 , openssl
+# The primary --enable-XXX variant. 'all' enables most features, but causes build-errors for some software,
+# requiring to build a special variant for that software. Example: 'haproxy'
+, variant ? "all"
+, extraConfigureFlags ? []
+, enableLto ? true
 }:
-
-stdenv.mkDerivation rec {
-  pname = "wolfssl";
+stdenv.mkDerivation (finalAttrs: {
+  pname = "wolfssl-${variant}";
   version = "5.6.3";
 
   src = fetchFromGitHub {
     owner = "wolfSSL";
     repo = "wolfssl";
-    rev = "refs/tags/v${version}-stable";
+    rev = "refs/tags/v5.6.3-stable";
```

Comment:
Why?

Reply:
Ahh, goofy mistake - Sandro mentioned that taking the finalAttrs.version  was silly - changing it would change the ref, causing the hash to mismatch.

I, in my infinite wisdom, just embedded the version.

But I can just make the thing rec as well and use version like before.

Reply:
Yeah, there's no harm in using `finalAttrs` it's just not a complete solution because nothing will ever be able to sort out the hash situation.

Reply:
OK, I reverted the change I made based on Sandro's review.
However, I am not sure if there is a way to satisfy you both when it comes to this.

Reply:
I'd just leave it as you found it - there's no perfect way to deal with this situation.

Reply:
Do you mean making the attr `rec` again and leaving just `version` in it? Or `finalAttrs.version` like it is now (once again)?

Reply:
Either frankly.